src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -77,12 +78,42 @@ static ss::sstring describe_as_string(const T& t) {
     return ssx::sformat("{}", t);
 }
 
+
+// property_config_type maps the datatype for a configuration to describe_configs_type.
+// Currently class_type and password are not used in Redpanda so we do not include checks for those types.
+// You may find a similar mapping in Apache Kafka at https://github.com/apache/kafka/blob/be032735b39360df1a6de1a7feea8b4336e5bcc0/core/src/main/scala/kafka/server/ConfigHelper.scala
+// Type sizes are based on Java and taken from https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html
+template<typename T>
+describe_configs_type property_config_type() {
+  // Default type is unknown
+  describe_configs_type type = describe_configs_type::unknown;
+
+  if constexpr (std::is_same_v<T, bool>) {
+    type = describe_configs_type::boolean;
+  } else if constexpr (std::is_same_v<T, ss::sstring>) {
+    type = describe_configs_type::string;
+  }else if constexpr (std::is_same_v<T, int32_t>) {
+    type = describe_configs_type::int_type;
+  }else if constexpr (std::is_same_v<T, int16_t>) {
+    type = describe_configs_type::short_type;
+  }else if constexpr (std::is_same_v<T, int64_t>) {
+    type = describe_configs_type::long_type;
+  }else if constexpr (std::is_floating_point_v<T>) {
+    type = describe_configs_type::double_type;
+  }else if constexpr (std::is_same_v<T, std::vector<T>>) {
```

Comment:
This won't work, grep the project for `is_std_vector` for how we do this within our serialization frameworks